packages/authenticator/amplify_authenticator/test/sign_up_form_test.dart

```diff
@@ -0,0 +1,168 @@
+
+          final passwordConfirmationFieldError = find.descendant(
+            of: signUpPage.confirmPasswordField,
+            matching: find.text('Confirm Password field must not be blank.'),
+          );
+
+          expect(passwordConfirmationFieldError, findsOneWidget);
+        },
+      );
+
+      testWidgets(
+        'displays message when submitted with invalid email address',
+        (tester) async {
+          await tester.pumpWidget(
+            const MockAuthenticatorApp(initialStep: AuthenticatorStep.signUp),
+          );
+          await tester.pumpAndSettle();
+
+          final signUpPage = SignUpPage(tester: tester);
+
+          await signUpPage.enterUsername('not-an-email');
+          await signUpPage.enterPassword('Password123');
+          await signUpPage.enterPasswordConfirmation('Password123');
+
+          await signUpPage.submitSignUp();
+
+          final usernameFieldError = find.descendant(
+            of: signUpPage.usernameField,
+            matching: find.text('Invalid email format.'),
+          );
+
+          expect(usernameFieldError, findsOneWidget);
+        },
+      );
+
+      testWidgets(
+        'displays message when password does not meet requirements',
+        (tester) async {
+          await tester.pumpWidget(
+            MockAuthenticatorApp(
+              initialStep: AuthenticatorStep.signUp,
+              config: jsonEncode(passwordReqConfig.toJson()),
+            ),
+          );
+          await tester.pumpAndSettle();
+
+          final signUpPage = SignUpPage(tester: tester);
+
+          await signUpPage.enterUsername('user@example.com');
+          await signUpPage.enterPassword('1234');
+
+          await signUpPage.submitSignUp();
+
+          final passwordFieldErrorLine1 = find.descendant(
+            of: signUpPage.passwordField,
+            matching: find.textContaining(
+              'Password must include',
+            ),
+          );
+
+          final passwordFieldErrorLine2 = find.descendant(
+            of: signUpPage.passwordField,
+            matching: find.textContaining(
+              '* at least 16 characters',
+            ),
+          );
+
+          final passwordFieldErrorLine3 = find.descendant(
+            of: signUpPage.passwordField,
+            matching: find.textContaining(
+              '* at least 1 uppercase character',
+            ),
+          );
+
+          final passwordFieldErrorLine4 = find.descendant(
+            of: signUpPage.passwordField,
+            matching: find.textContaining(
+              '* at least 1 symbol character',
+            ),
+          );
+
+          expect(passwordFieldErrorLine1, findsOneWidget);
+          expect(passwordFieldErrorLine2, findsOneWidget);
+          expect(passwordFieldErrorLine3, findsOneWidget);
+          expect(passwordFieldErrorLine4, findsOneWidget);
+        },
+      );
+    });
+  });
+}
+
+const passwordReqConfig = AmplifyConfig(
+  auth: AuthConfig(
```

Comment:
For future ref, you can use `AuthConfig.cognito` to make this easier

Reply:
Oh, nice! I will update this